### PR TITLE
 show call cache at the submission level. Hide some call details for subworkflows

### PR DESCRIPTION
**Pull request by davidangb**
## Dear Reviewer:

you probably want to 1) review each commit individually, and 2) ignore whitespace when viewing the diff.

The first commit:
* displays the user's choice of enable/disable call cache at the submission level, beside other submission details
* removes the call-cache display from individual workflows
* hides stdout/stderr lines for calls that are subworkflows, since those values don't exist at this level (you have to query the subworkflow directly)

The second commit:
* changes the submission details from a two-column display to three columns. This includes an indentation change, moving some lines of code around, and causes huge diffs in github.

Third commit:
* a more generic way to show/hide some call fields - hide if the field is nil.

More commits:
* bonus feature! Display the workflow's status text next to its icon (GAWB-2098)
* bonus feature! Hover text over each submission's status in the main monitor tab shows summary of workflow statuses. (kinda GAWB-2355)
* bonus feature! Display a summary of call statuses next to unexpanded calls (GAWB-873, GAWB-223)
* bonus feature! More consistent icons for submission status on the monitor tab. Previously, many rows had no icon: Submitted, Aborting. This also eliminates a JS error in console.

### Before
submission-details-before

### After
submission-details-after

### Before
workflow-details-before
_(the root issue this ticket is about is that the call cache values displayed here are incorrect. I'm solving it by moving the call cache value to the submission level.)_

### After
workflow-details-after

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/page/workspace/monitor/workflow_details.cljs [hunk 1]
```diff
@@ -286,21 +284,23 @@
...
                 (create-field "Started" (moncommon/render-date (data "start")))
                 ;(utils/cljslog data)
                 (create-field "Ended" (moncommon/render-date (data "end")))
                 [IODetail {:label "Inputs" :data (data "inputs") :call-detail? true}]
                 [IODetail {:label "Outputs" :data (data "outputs") :call-detail? true}]
-                (create-field "stdout" (display-value (data "stdout") (last (string/split (data "stdout") #"/"))))
-                (create-field "stderr" (display-value (data "stderr") (last (string/split (data "stderr") #"/"))))
+                (when (data "stdout")
+                  (create-field "stdout" (display-value (data "stdout") (last (string/split (data "stdout") #"/")))))
+                (when (data "stdout")
```
Comment (ghost) [inline src/cljs/main/broadfcui/page/workspace/monitor/workflow_details.cljs]: did you mean `(data "stderr")` here?

**Review by ghost — APPROVED** [review]
One comment, wasn't able to test. Looks good! 👍

**Comment by davidangb** [comment]
One Selenium test failure in https://fc-jenkins.dsp-techops.broadinstitute.org/job/firecloud-ui-fiab-test-runner/3928/consoleFull:
```
[info] *** 1 TEST FAILED ***
[error] Failed: Total 124, Failed 1, Errors 0, Passed 123, Ignored 3
[error] Failed tests:
[error] 	org.broadinstitute.dsde.firecloud.test.user.SignInSpec

[info]   - should be able to log in and out multiple times as multiple users *** FAILED *** (1 minute, 35 seconds)
[info]     org.openqa.selenium.TimeoutException: Expected condition failed: waiting for org.broadinstitute.dsde.workbench.service.test.WebBrowserUtil$await$$anon$2@4b023188 (tried for 40 second(s) with 500 milliseconds interval)
```

**Comment by davidangb** [comment]
`SignInSpec` passed when run in a manual swatomation pipeline: https://fc-jenkins.dsp-techops.broadinstitute.org/view/Testing/view/Pipeline%20Jobs/job/swatomation-pipeline/6953/

File: src/cljs/main/broadfcui/page/workspace/monitor/workflow_details.cljs [hunk 2]
```diff
@@ -286,21 +284,23 @@
...
                 (create-field "Started" (moncommon/render-date (data "start")))
                 ;(utils/cljslog data)
                 (create-field "Ended" (moncommon/render-date (data "end")))
                 [IODetail {:label "Inputs" :data (data "inputs") :call-detail? true}]
                 [IODetail {:label "Outputs" :data (data "outputs") :call-detail? true}]
-                (create-field "stdout" (display-value (data "stdout") (last (string/split (data "stdout") #"/"))))
-                (create-field "stderr" (display-value (data "stderr") (last (string/split (data "stderr") #"/"))))
+                (when (not (data "subWorkflowId"))
+                  (do
+                    (create-field "stdout" (display-value (data "stdout") (last (string/split (data "stdout") #"/"))))
```
Comment (jmthibault79) [inline src/cljs/main/broadfcui/page/workspace/monitor/workflow_details.cljs]: I'd prefer a more generic "when data exists" like the `when-let`s around 315

Comment (jmthibault79) [reply src/cljs/main/broadfcui/page/workspace/monitor/workflow_details.cljs]: oh you did that on the next commit :)

Then I want to update this by saying to use when-let so we don't need to reference the data twice

File: src/cljs/main/broadfcui/page/workspace/monitor/tab.cljs [hunk 3]
```diff
@@ -35,7 +35,9 @@
        :render #(links/create-internal {:data-test-id (str "submission-" %)
                                         :href (nav/get-link :workspace-submission workspace-id %)}
                   "View")}
-      {:header "Status" :as-text :status :sort-by :text
+      {:header "Status"
+       :sort-by (fn [submission] (:status submission))
+       :as-text (fn [submission] (.stringify js/JSON (clj->js (:workflowStatuses submission))))
```
Comment (jmthibault79) [inline src/cljs/main/broadfcui/page/workspace/monitor/tab.cljs:40]: Dunno how users will respond to JSON - but it works great as a first iteration.  👍

**Review by jmthibault79 — APPROVED** [review]

**Review by ghost — APPROVED** [review]
👍

**Pull request merged**